### PR TITLE
Bump CDk

### PR DESCRIPTION
**Pull request by benmoriceau**
## What
Bump in order to include the reduction of the async buffer block size

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/resources/pull-requests-handbook)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:generateScaffolds` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

<details><summary><strong>Updating the Python CDK</strong></summary>

### Airbyter

Before merging:
- [ ] Pull Request description explains what problem it is solving
- [ ] Code change is unit tested
- [ ] Build and my-py check pass
- [ ] Smoke test the change on at least one affected connector
   - On Github: Run [this workflow](https://github.com/airbytehq/airbyte/actions/workflows/connectors_tests.yml), passing `--use-local-cdk --name=source-<connector>` as options
   - Locally: `airbyte-ci connectors --use-local-cdk --name=source-<connector> test`
- [ ] PR is reviewed and approved
      
After merging:
- [ ] [Publish the CDK](https://github.com/airbytehq/airbyte/actions/workflows/publish-cdk-command-manually.yml)
   - The CDK does not follow proper semantic versioning. Choose minor if this the change has significant user impact or is a breaking change. Choose patch otherwise.
   - Write a thoughtful changelog message so we know what was updated.
- [ ] Merge the platform PR that was auto-created for updating the Connector Builder's CDK version
   - This step is optional if the change does not affect the connector builder or declarative connectors.

</details>

### CODE REVIEW

**Review by edgao — APPROVED** [review]

File: airbyte-cdk/java/airbyte-cdk/README.md [hunk 1]
```diff
@@ -154,9 +154,10 @@ MavenLocal debugging steps:
...
-| :------ | :--------- | :--------------------------------------------------------- | :------------------------------------ |
+| Version | Date       | Pull Request                                               | Subject                                                                                                                                                        |
+|:--------|:-----------| :--------------------------------------------------------- |:---------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 0.1.4   | 2023-10-06 | [\#31139](https://github.com/airbytehq/airbyte/pull/31139) | Reduce async buffer                                                                                                                                            |
 | 0.1.1   | 2023-09-28 | [\#30835](https://github.com/airbytehq/airbyte/pull/30835) | JDBC destinations now avoid staging area name collisions by using the raw table name as the stage name. (previously we used the stream name as the stage name) |
```
Comment (evantahler) [inline airbyte-cdk/java/airbyte-cdk/README.md:160]: Looks like we missed some versions here :/

File: airbyte-integrations/connectors/destination-redshift/build.gradle [hunk 2]
```diff
@@ -4,7 +4,7 @@ plugins {
 }
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-redshift/build.gradle:4]: I think we are missing a redshift metadata version bump

**Review by evantahler — CHANGES_REQUESTED** [review]

File: airbyte-cdk/java/airbyte-cdk/README.md (hunk 1 above)
Comment (benmoriceau) [reply airbyte-cdk/java/airbyte-cdk/README.md:160]: Yep

File: airbyte-integrations/connectors/destination-redshift/build.gradle (hunk 2 above)
Comment (benmoriceau) [reply airbyte-integrations/connectors/destination-redshift/build.gradle:4]: I think that the build is not fixed yet so I didn't add it on purpose.

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`256caa0290`](https://github.com/airbytehq/airbyte/pull/31139/commits/256caa02900a369acf470afa274bf747811803f0)) - ❌

⏲️  Total pipeline duration: 01mn45s 

| Step                                        | Result   |
|:--------------------------------------------|:---------|
| Build connector tar                         | ❌        |
| Validate metadata for destination-snowflake | ✅        |
| Connector version semver check              | ✅        |
| Connector version increment check           | ✅        |
| QA checks                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/bmoric_bump-cdk/1696624675/256caa02900a369acf470afa274bf747811803f0/destination-snowflake/3.1.21/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:256caa02900a369acf470afa274bf747811803f0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`256caa0290`](https://github.com/airbytehq/airbyte/pull/31139/commits/256caa02900a369acf470afa274bf747811803f0)) - ❌

⏲️  Total pipeline duration: 01mn35s 

| Step                                       | Result   |
|:-------------------------------------------|:---------|
| Build connector tar                        | ❌        |
| Validate metadata for destination-bigquery | ✅        |
| Connector version semver check             | ✅        |
| Connector version increment check          | ✅        |
| QA checks                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/bmoric_bump-cdk/1696624675/256caa02900a369acf470afa274bf747811803f0/destination-bigquery/2.0.24/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:256caa02900a369acf470afa274bf747811803f0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-redshift test report (commit [`256caa0290`](https://github.com/airbytehq/airbyte/pull/31139/commits/256caa02900a369acf470afa274bf747811803f0)) - ❌

⏲️  Total pipeline duration: 46.39s 

| Step                                       | Result   |
|:-------------------------------------------|:---------|
| Build connector tar                        | ❌        |
| Validate metadata for destination-redshift | ✅        |
| Connector version semver check             | ✅        |
| QA checks                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/bmoric_bump-cdk/1696624675/256caa02900a369acf470afa274bf747811803f0/destination-redshift/0.6.6/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:256caa02900a369acf470afa274bf747811803f0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-redshift test
```

File: airbyte-integrations/connectors/destination-redshift/build.gradle (hunk 2 above)
Comment (benmoriceau) [reply airbyte-integrations/connectors/destination-redshift/build.gradle:4]: Redshift was not publish with the 30 MB bump because of the build issue

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`256caa0290`](https://github.com/airbytehq/airbyte/pull/31139/commits/256caa02900a369acf470afa274bf747811803f0)) - ❌

⏲️  Total pipeline duration: 08mn25s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Build connector tar                                                  | ✅        |
| Build destination-bigquery docker image for platform(s) linux/x86_64 | ✅        |
| Java Connector Unit Tests                                            | ❌        |
| Java Connector Integration Tests                                     | ✅        |
| Validate metadata for destination-bigquery                           | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/bmoric_bump-cdk/1696626092/256caa02900a369acf470afa274bf747811803f0/destination-bigquery/2.0.24/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:256caa02900a369acf470afa274bf747811803f0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`efc926ba17`](https://github.com/airbytehq/airbyte/pull/31139/commits/efc926ba1712c433356edb46c4a1f3fa4973057a)) - ✅

⏲️  Total pipeline duration: 10mn58s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Build connector tar                                                   | ✅        |
| Build destination-snowflake docker image for platform(s) linux/x86_64 | ✅        |
| Java Connector Unit Tests                                             | ✅        |
| Java Connector Integration Tests                                      | ✅        |
| Validate metadata for destination-snowflake                           | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/bmoric_bump-cdk/1696626896/efc926ba1712c433356edb46c4a1f3fa4973057a/destination-snowflake/3.1.21/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:efc926ba1712c433356edb46c4a1f3fa4973057a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-redshift test report (commit [`efc926ba17`](https://github.com/airbytehq/airbyte/pull/31139/commits/efc926ba1712c433356edb46c4a1f3fa4973057a)) - ❌

⏲️  Total pipeline duration: 02mn41s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Build connector tar                                                  | ✅        |
| Build destination-redshift docker image for platform(s) linux/x86_64 | ❌        |
| Validate metadata for destination-redshift                           | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/bmoric_bump-cdk/1696626896/efc926ba1712c433356edb46c4a1f3fa4973057a/destination-redshift/0.6.6/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:efc926ba1712c433356edb46c4a1f3fa4973057a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-redshift test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`efc926ba17`](https://github.com/airbytehq/airbyte/pull/31139/commits/efc926ba1712c433356edb46c4a1f3fa4973057a)) - ✅

⏲️  Total pipeline duration: 07mn38s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Build connector tar                                                  | ✅        |
| Build destination-bigquery docker image for platform(s) linux/x86_64 | ✅        |
| Java Connector Unit Tests                                            | ✅        |
| Java Connector Integration Tests                                     | ✅        |
| Validate metadata for destination-bigquery                           | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/bmoric_bump-cdk/1696626896/efc926ba1712c433356edb46c4a1f3fa4973057a/destination-bigquery/2.0.24/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:efc926ba1712c433356edb46c4a1f3fa4973057a)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-approvington** [comment]
You did it! <br/> finish line

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**